system/jlib/jdebug.hpp

```diff
@@ -225,6 +225,43 @@ class jlib_decl HiresTimer
     unsigned start;
 };
 
+//---------------------------------------------------------------------------------------------------------------------
+
+/*
+  There are many situations where you want to report or check something periodically.  This class encapsulates th code
+  to perform the check.
+  The constructor has an option to indicate whether the first hasElapsed call should return true, or be suppressed.
+*/
+class PeriodicTimer
+{
+public:
+    PeriodicTimer(unsigned ms, bool suppressFirst)
+    {
+        timePeriodCycles = nanosec_to_cycle(ms * 1000000);
```

Comment:
ms * 1000000 will overflow, think needs changing to: (__int64)ms * 1000000

Reply:
well spotted